main.go

```diff
@@ -131,54 +182,45 @@ func releaseToStream(releaseArch *ReleaseArch, rel Release) StreamArch {
 	// }
 
 	if releaseArch.Media.Openstack != nil {
-		openstack := StreamMediaDetails{
+		artifacts["openstack"] = stream.PlatformArtifacts{
 			Release: rel.Release,
-			Formats: releaseArch.Media.Openstack.Artifacts,
+			Formats: mapFormats(releaseArch.Media.Openstack.Artifacts),
 		}
-		artifacts.Openstack = &openstack
 	}
 
 	if releaseArch.Media.Qemu != nil {
-		qemu := StreamMediaDetails{
+		artifacts["qemu"] = stream.PlatformArtifacts{
 			Release: rel.Release,
-			Formats: releaseArch.Media.Qemu.Artifacts,
+			Formats: mapFormats(releaseArch.Media.Qemu.Artifacts),
 		}
-		artifacts.Qemu = &qemu
 	}
 
-	if releaseArch.Media.Virtualbox != nil {
-		virtualbox := StreamMediaDetails{
-			Release: rel.Release,
-			Formats: releaseArch.Media.Virtualbox.Artifacts,
-		}
-		artifacts.Virtualbox = &virtualbox
-	}
+	// if releaseArch.Media.Virtualbox != nil {
+	// 	virtualbox := stream.PlatformArtifacts{
+	// 		Release: rel.Release,
+	// 		Formats: releaseArch.Media.Virtualbox.Artifacts,
+	// 	}
+	// 	artifacts.Virtualbox = &virtualbox
+	// }
```

Comment:
Ah right I meant to hoist out the virtualbox disablement too.  Eh.  I can if you guys care.

Reply:
A separate commit would be nice, if convenient.  A separate PR isn't necessary.

Reply:
OK did that and also a separate commit for the url function.